tools/autograd/gen_inplace_or_view_type.py

```diff
@@ -170,13 +173,10 @@ def unpack_args(f: NativeFunction) -> Tuple[List[str], List[Binding]]:
             continue
 
         is_tensor_list = is_tensor_list_type(binding.argument.type)
-        ref = (not is_nullable) and not is_tensor_list
-        suffix = '_opt' if is_nullable and not is_tensor_list else ''
-        body.append(UNPACK_TENSOR.substitute(
+        template = (UNPACK_TENSOR_LIST if is_tensor_list else UNPACK_TENSOR)
```

Comment:
I'm not sure how this is equivalent to the old code, in particular, I don't see the `_list` magic there.
Did you diff the generated code to make sure it is still the same?

Reply:
I didn't. (I'll make a note to do so.) list magic is now handled by using a different codegen template. The key point for understanding equivalence is that there are no instances of unpack_opt, so the logic simplifies.